plugins/exporter.koplugin/formats/joplin.lua

```diff
@@ -0,0 +1,238 @@
+local BD = require("ui/bidi")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local T = require("ffi/util").template
+local _ = require("gettext")
+local JoplinClient = require("clients/JoplinClient")
+
+local JoplinExporter = require("formats/base"):new {
+    name = "joplin",
+    is_remote = true,
+    version = "1.0.0"
+}
+
+function JoplinExporter:isEnabled()
+    return self.settings.enabled and self.settings.ip and self.settings.port and self.settings.token
+end
+
+function JoplinExporter:toggleEnabled()
+    if not self.settings.ip or not self.settings.port or not self.settings.token then return end
+    self.settings.enabled = not self.settings.enabled
+    self:saveSettings()
+end
+
+function JoplinExporter:init()
```

Comment:
this isn't  intended to be used that way. the BaseExporter:init() function is called when a new object is created and should return the object instead:

```lua
function BaseExporter:new(o)
    o = o or {}
    o.id = "exporter"
    o.name = o.name or "generic"
    o.extension = o.extension or o.name or "export"
    o.version = o.version or "1.0.0"
    o.clipping_dir = DataStorage:getDataDir() .. "/clipboard"
    o.is_remote = o.is_remote or false
    setmetatable(o, self)
    self.__index = self
    return o:init()
end

function BaseExporter:init()
    self:loadSettings()
    return self
end
```

So, if you think *some* of the formats will require a different init stuff  after loading settings then factorize that too. (ie: create an empty BaseExporter:pluginInit() and change the function to call that after loadSettings.

Then on the format you need init just implement MyExporter:pluginInit() and do there stuff that belongs to the plugin

Reply:
Done.